src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -53,5 +63,51 @@ public void saveDebtPosition(DebtPositionDTO debtPositionDTO) {
       });
     });
   }
+
+  @Override
+  public void finalizeSyncStatus(Long debtPositionId, Map<String, IudSyncStatusUpdateDTO> syncStatusDTO) {
+    DebtPosition debtPosition = debtPositionRepository.findOneWithAllDataByDebtPositionId(debtPositionId);
+
+    debtPosition.getPaymentOptions().forEach(paymentOption ->
+      paymentOption.getInstallments().stream()
+        .filter(installment -> TO_SYNC.name().equals(installment.getStatus()))
+        .filter(installment -> syncStatusDTO.containsKey(installment.getIud()))
+        .filter(installment -> installment.getIupdPagopa().equals(syncStatusDTO.get(installment.getIud()).getIupdPagopa()))
```

Comment:
the iupdPagopa on the request represent the value to store on the db! not a field to check if match ;)
(we are storing the nodo synchronization result, whose will provide us the pagopaIds